### ISSUE TITLE
PUT Mapping enforces _timestamp to be set every time if index has it stored

### ISSUE THREAD

**Issue by andrestc** [elastic/elasticsearch#13472]
This was found on ES 1.7.1.

Creating an index with _timestamp enabled:
```
curl -XPOST localhost:9200/test -d '{
    "settings" : {
        "number_of_shards" : 1
    },
    "mappings" : {
        "type1" : {
            "_timestamp": {"enabled": true, "store": true},
            "properties" : {
                "field1" : { "type" : "string", "index" : "not_analyzed" }
            }
        }
    }
}'
```

If we try creating a new field, with PUT Mapping API:
```
curl -XPUT 'http://localhost:9200/test/_mapping/type1' -d '
{
  "type1": {
    "properties": {
      "field2": {"type": "string"}
    }
  }
}
'
```

It fails saying _timestamp has different store values:
```
[2015-09-10 11:27:45,636][DEBUG][action.admin.indices.mapping.put] [Black Tom Cassidy] failed to put mappings on indices [[test]], type [type1]
org.elasticsearch.index.mapper.MergeMappingException: Merge failed with failures {[mapper [_timestamp] has different store values]}
	at org.elasticsearch.cluster.metadata.MetaDataMappingService$4.execute(MetaDataMappingService.java:511)
	at org.elasticsearch.cluster.service.InternalClusterService$UpdateTask.run(InternalClusterService.java:374)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:188)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:158)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

To be able to create the new mapping, we need to specify the _timestamp configuration again:
```
curl -XPUT 'http://localhost:9200/test/_mapping/type1' -d '
{
  "type1": {
    "_timestamp" : { "enabled" : true, "store": "true" },
    "properties": {
      "field2": {"type": "string"}
    }
  }
}
'
```

Unless I'm missing something, I believe this is not de desired behavior. I think since I'm not stating anything about the _timestamp field on the Put Mapping, it should succeed keeping the configuration stated on index creation time.

If this is indeed a bug, I can try submiting a PR for it.

**Comment by eskibars** [comment]
I'm able to reproduce.  If not defined, it appears the _timestamp type is implied (and implied to have the default value of "false").  So, adding ?ignore_conflicts=true also allows the update mapping to go through, but can have other (undesired) consequences if you had a "real" conflict.  So yes, looks like a bug.  _timestamp is deprecated in 2.0 though, in favor of using a normal date field, so it may be best to treat this as a limitation in 1.x, since there is a (relatively simple) workaround.  Thoughts?

**Comment by clintongormley** [comment]
This has been fixed in 2.0
